sdcm/cluster.py

```diff
@@ -4082,11 +4084,37 @@ def stop_nemesis(self, timeout=10):
         if self.nemesis_termination_event.is_set():
             return
         self.log.info('Set _nemesis_termination_event')
+        self.log.debug("There are %s nemesis threads currently running", len(self.nemesis_threads))
         self.nemesis_termination_event.set()
+        threads_tracebacks = []
+        # pylint: disable=protected-access
+        current_thread_frames = sys._current_frames()
         for nemesis_thread in self.nemesis_threads:
             nemesis_thread.join(timeout)
+            if nemesis_thread.is_alive():
+                stack_trace = traceback.format_stack(current_thread_frames[nemesis_thread.ident])
+                threads_tracebacks.append("\n".join(stack_trace))
+        self.argus_terminate_remaining_nemeses(stack_trace="\n".join(threads_tracebacks))
         self.nemesis_threads = []
 
+    def argus_terminate_remaining_nemeses(self, stack_trace="") -> None:
+        try:
+            run = ArgusTestRun.get()
+            running_nemeses = [
+                nemesis
+                for nemesis in run.run_info.results.nemesis_data
```

Comment:
we use list comprehension when it fits one line...

Reply:
corrected